quesma/queryparser/pancake_transformer.go

```diff
@@ -297,21 +300,23 @@ func (a *pancakeTransformer) findParentBucketLayer(layers []*pancakeModelLayer,
 	return layer, nil
 }
 
-func (a *pancakeTransformer) createTopHitPancakes(pancake *pancakeModel) (result []*pancakeModel, err error) {
+func (a *pancakeTransformer) createTopHitAndTopMetricsPancakes(pancake *pancakeModel) (result []*pancakeModel, err error) {
 	for layerIdx, layer := range pancake.layers {
 		metricsWithoutTopHits := make([]*pancakeModelMetricAggregation, 0, len(layer.currentMetricAggregations))
-		for _, metric := range layer.currentMetricAggregations {
-			if _, isTopHits := metric.queryType.(metrics_aggregations.TopHits); isTopHits {
-				canOptimize := layerIdx == len(pancake.layers)-1 && len(layer.currentMetricAggregations) == 1
+		for metricIDx, metric := range layer.currentMetricAggregations {
```

Comment:
```suggestion
		for metricIdx, metric := range layer.currentMetricAggregations {
```
to be consistent with naming everywhere

Reply:
Fixed.